### PR TITLE
internal: Migrate to gitserver.NewClient

### PR DESCRIPTION
**Pull request by indradhanush**
Continuation of work started in #54788. Some more to follow in a separate PR.

Primarily migrated usages in internal and enterprise/internal packages and as a result the code changes that follow in other packages are related to propagate database.DB/gitserver.Client.

Also changes to internal/repos as well to do the migration.

Part of #54741.

## Test plan

- Played around locally, did not spot any regressions
- CI

### CODE REVIEW

**Review by eseliger — APPROVED** [review]

File: internal/compute/command.go [hunk 1]
```diff
@@ -13,10 +13,16 @@ type Command interface {
...
+type CommandPostRunHook interface {
+	PostRunHook(*commandParser)
+}
+
 var (
 	_ Command = (*MatchOnly)(nil)
 	_ Command = (*Replace)(nil)
 	_ Command = (*Output)(nil)
+
+	_ CommandPostRunHook = (*Replace)(nil)
```
Comment (keegancsmith) [inline internal/compute/command.go:25]: this seems a bit dodgy to me to be honest.
1. Parsing a string now requires a gitserver client.
2. You depend on a hook to set a gitserver client.
  - This seems fragile
  - You need it to do Run. Surely it is a PreRunHook?

Would this change be smaller and less fragile if instead you updated Command.Run to take a gitserver client? Or maybe a struct of runtime dependencies?

Comment (indradhanush) [reply internal/compute/command.go]: @keegancsmith Agreed. I wasn't liking it a lot myself and didn't think of it this way as a possible alternative.

This is a good call out. I updated the design so now commands can be configured with an "option". I was trying to patch this to update `Command.Run` initially but then the approach of adding option on the command itself felt cleaner.

Could you please take another look?

**Comment by indradhanush** [comment]
Oh no, I've been conflicted. 😢 
Will rebase and update the PR and re-request reviews.
image

**Comment by indradhanush** [comment]
Okay the conflicts weren't too bad. Mostly import path related.

File: internal/compute/command.go (hunk 1 above)
Comment (keegancsmith) [reply internal/compute/command.go]: I'll take a deeper look today. It still feels wrong to me that Parsing a string requires a gitserver client. Maybe the issue is it just has the wrong name and it isn't parsing. But will get back to you today after a much deeper look.

Comment (indradhanush) [reply internal/compute/command.go]: Happy to get on a call and pair if you think that would be better.

Comment (keegancsmith) [reply internal/compute/command.go]: @indradhanush I was looking at how run worked in my editor and got carried away and made it take in a gitserver client. I pushed a bunch of commits seperating out the separate steps. I think this is a smaller change and makes more sense from the perspective of parsing vs running.

Comment (indradhanush) [reply internal/compute/command.go]: Taking a look now.

Comment (indradhanush) [reply internal/compute/command.go]: Ah this is much better! Thank you!

**Comment by indradhanush** [comment]
@keegancsmith Shall I merge?

**Comment by keegancsmith** [comment]
go for it before Eric or Camden do another huge refactor.

**Pull request merged**